power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -598,17 +598,130 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
     static constexpr auto transformer_index_of = container_impl::get_cls_pos_v<T, TransformerTypes...>;
     static_assert(((transformer_index_of<TransformerTypes> < sizeof...(TransformerTypes)) && ...));
 
+    struct BinarySearch {
+      public:
+        BinarySearch() = default;
+        BinarySearch(IntS tap_pos, IntS tap_min, IntS tap_max) { reset_bs(tap_pos, tap_min, tap_max); }
+
+        IntS get_bs_current_tap() const { return current; }
+        bool get_bs_last_down() const { return last_down; }
+        bool get_bs_inevitable_run() const { return inevitable_run; }
+        bool get_end_of_bs() const { return get_bs_tap_lower_bound() >= get_bs_tap_upper_bound(); }
+
+        void set_bs_current_tap(IntS _current_tap) { this->current = _current_tap; }
+        void set_bs_last_check(bool _last_check) { this->last_check = _last_check; }
+        void set_bs_inevitable_run(bool _inevitable_run) { this->inevitable_run = _inevitable_run; }
+
+        void recalibrate(bool strategy_max) {
+            if (get_bs_tap_reverse() ? strategy_max : !strategy_max) {
```

Comment:
```suggestion
            if (get_bs_tap_reverse() == strategy_max) {
```
or maybe use an intermediate variable

Reply:
comment addressed in the latest commit.